### PR TITLE
Fixed a case with empty order in transpose op

### PR DESCRIPTION
**Pull request by ilyachur**
Task: 32756

### CODE REVIEW

**Comment by mikhail-treskin** [comment]
Can you please remove the skip pattern for this case in single layer tests:
https://github.com/openvinotoolkit/openvino/blob/a4c1c6fc4641dad41d9d27459c197260d7b55f35/inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp#L26
Most likely test skipped for other plugins, can you please search in tests by the issue number and remove all the skip patterns related to the fixed issue.

**Comment by ilyachur** [comment]
> Can you please remove the skip pattern for this case in single layer tests:
> https://github.com/openvinotoolkit/openvino/blob/a4c1c6fc4641dad41d9d27459c197260d7b55f35/inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp#L26
> 
> 
> Most likely test skipped for other plugins, can you please search in tests by the issue number and remove all the skip patterns related to the fixed issue.

Removed

**Review by dmitry-gorokhov — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_permute_node.cpp [hunk 1]
```diff
@@ -148,6 +148,13 @@ void MKLDNNPermuteNode::getSupportedDescriptors() {
     std::vector<int> layerOrder = layer->GetParamAsInts("order");
     for (auto ord : layerOrder)
         order.push_back(static_cast<size_t>(ord));
+
+    if (order.empty()) {
```
Comment (lazarevevgeny) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_permute_node.cpp:152]: You fixed the CPU plugin. But what about others?

Comment (ilyachur) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_permute_node.cpp:152]: At the current we have such tests only for CPU.
I think if these tests will be extended for other plugins, plugin owners will fix implementations.

File: ngraph/core/src/op/transpose.cpp [hunk 2]
```diff
@@ -50,14 +50,21 @@ void op::v1::Transpose::validate_and_infer_types()
 
     const auto& arg_shape = get_input_partial_shape(0);
     NODE_VALIDATION_CHECK(this,
-                          input_order_shape.compatible(PartialShape{arg_shape.rank()}),
+                          input_order_shape.compatible(PartialShape{arg_shape.rank()}) ||
+                              (input_order_shape.is_static() && input_order_shape.rank() == 1 &&
```
Comment (lazarevevgeny) [inline ngraph/core/src/op/transpose.cpp:54]: Why do you added this check for `input_order_shape[0] == 0`? I do understand that if the input tensor rank is 1 then it is the only correct permutation. But if you want to check a permutation then it should be done separately and for all valid input ranks

Comment (ilyachur) [reply ngraph/core/src/op/transpose.cpp:54]: The problem that in order to create a empty constant we need to create something like that:
```
Constant(Shape{0}, {} /*data*/)
```
This order shape is incompatible with data shape, but we know that it is a special case when we need to reverse input shapes order. Exactly this case I checked here.

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**